rpc2/dispatch.go

```diff
@@ -6,9 +6,13 @@ type DecodeNext func(interface{}) error
 type ServeHook func(DecodeNext) (interface{}, error)
 
 type Dispatcher interface {
-	Dispatch(m *Message) error
 	Call(name string, arg interface{}, res interface{}, f UnwrapErrorFunc) error
 	RegisterProtocol(Protocol) error
+}
+
+type dispatcher interface {
+	Dispatcher
```

Comment:
We can wrap the `Server` and `Client` functionality directly into `Transporter`, in which case dispatcher can be package local. Otherwise, `Transporter.GetDispatcher` is needed (even though it's only used inside rpc2's `Server` and `Client` constructs).

My preference would be to completely include the `Server` and `Client` APIs in `Transporter`, which would then look like:
```go
type Transporter interface {
    IsConnected() bool
    RegisterProtocol(Protocol) error
    RunCall(method string, arg interface{}, res interface{}, f UnwrapErrorFunc) error
    RunServer() error
}
```

This way the `Transporter` only includes functionality that has external relevance, and `Client`/`Server` then only serve the purpose of providing the correct `wrap`/`unwrap` functions.

WDYT @maxtaco?